### ISSUE TITLE
rmagick installation issue after a brew uprage

### ISSUE THREAD

**Issue by solisoft** [rmagick-temp/rmagick#60]
I have trouble after a brew update && brew upgrade

Here the log
Gem::Installer::ExtensionBuildError: ERROR: Failed to build gem native extension.

/Users/olivier/.rvm/rubies/ruby-1.9.3-p194/bin/ruby extconf.rb 
checking for Ruby version >= 1.8.5... yes
extconf.rb:128: Use RbConfig instead of obsolete and deprecated Config.
checking for clang... yes
checking for Magick-config... yes
checking for ImageMagick version >= 6.4.9... yes
checking for HDRI disabled version of ImageMagick... yes
checking for stdint.h... yes
checking for sys/types.h... yes
checking for wand/MagickWand.h... yes
checking for InitializeMagick() in -lMagickCore... no
checking for InitializeMagick() in -lMagick... no
checking for InitializeMagick() in -lMagick++... no
Can't install RMagick 2.13.1. Can't find the ImageMagick library or one of the dependent libraries. Check the mkmf.log file for more detailed information.

**\* extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of
necessary libraries and/or headers.  Check the mkmf.log file for more
details.  You may need configuration options.

Does anyone has the same issue ?

**Comment by timothyklim** [comment]
same issue

**Comment by solisoft** [comment]
I don't think it's a rmagick issue ... 
To fix it you'll have to downgrade imagemagick
https://gist.github.com/4336148

**Comment by ryansch** [comment]
Thanks for that gist.

**Comment by jedschneider** [comment]
the issue is that imagemagick has renamed the files that rmagick is looking for:

```
cd /usr/local/Cellar/imagemagick/6.8.0-10/lib/
ln -s libMagick++-Q16.7.dylib   libMagick++.dylib
ln -s libMagickCore-Q16.7.dylib libMagickCore.dylib
ln -s libMagickWand-Q16.7.dylib libMagickWand.dylib
gem install rmagick
```

**Comment by cromulus** [comment]
gem install rmagick fails with ubuntu 12.04.

`/usr/local/rvm/rubies/ruby-1.9.3-p194/bin/ruby extconf.rb
checking for Ruby version >= 1.8.5... yes
extconf.rb:128: Use RbConfig instead of obsolete and deprecated Config.
checking for gcc... yes
checking for Magick-config... no
Can't install RMagick 2.13.1. Can't find Magick-config in /usr/local/rvm/gems/ruby-1.9.3-p194/bin:/usr/local/rvm/gems/ruby-1.9.3-p194@global/bin:/usr/local/rvm/rubies/ruby-1.9.3-p194/bin:/usr/local/rvm/gems/ruby-1.9.3-p194/bin:/usr/local/rvm/gems/ruby-1.9.3-p194@global/bin:/usr/local/rvm/rubies/ruby-1.9.3-p194/bin:/usr/local/rvm/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games

**\* extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of
necessary libraries and/or headers.  Check the mkmf.log file for more
details.  You may need configuration options.

Provided configuration options:
    --with-opt-dir
    --with-opt-include
    --without-opt-include=${opt-dir}/include
    --with-opt-lib
    --without-opt-lib=${opt-dir}/lib
    --with-make-prog
    --without-make-prog
    --srcdir=.
    --curdir
    --ruby=/usr/local/rvm/rubies/ruby-1.9.3-p194/bin/ruby

Gem files will remain installed in /usr/local/rvm/gems/ruby-1.9.3-p194/gems/rmagick-2.13.1 for inspection.
Results logged to /usr/local/rvm/gems/ruby-1.9.3-p194/gems/rmagick-2.13.1/ext/RMagick/gem_make.out`

**Comment by arthurnn** [comment]
having some in here.. it looks like that rmagick doest no work with imagemagick 6.8 ... 

u will have to downgrade imagemagick to 6.7  till there is no fix on rmagick for it.

**Comment by jamilbk** [comment]
Thanks @jedschneider that worked. But I was still getting an error about a missing 64-bit libltdl which I fixed with:

``` bash
brew install libtool
brew link libtool
```

**Comment by djudd** [comment]
This workaround--from [here](https://coderwall.com/p/wnomjg)--works for me: 

``` bash
cd /usr/local/Cellar/imagemagick/6.8.0-10/lib
ln -s libMagick++-Q16.7.dylib   libMagick++.dylib
ln -s libMagickCore-Q16.7.dylib libMagickCore.dylib
ln -s libMagickWand-Q16.7.dylib libMagickWand.dylib
```

Is this project still active? If I submit a pull request is there any possibility it will get merged?

**Comment by thicksoup** [comment]
A simpler way to downgrade imagemagick with homebrew is:

brew switch imagemagick 6.7.7-6

**Comment by albandiguer** [comment]
thanks @jedschneider that did the job.

**Comment by caarlos0** [comment]
Same issue after update arch linux today.

**Comment by amatriain** [comment]
Same here after updating arch linux. Downgrading imagemagick package to 6.8.0.7-1 allows me to install rmagick again.

As stated above the problem is the name change in several dynamic link libraries (.so files). I'm not sure if this should be considered a bug in imagemagick or rmagick. Maybe the easiest solution would be that the arch package provided symbolic links with the old file names.

**Comment by amatriain** [comment]
According to an Arch Linux dev, rmagick should improve its imagemagick detection routines, so that it can find the imagemagick libraries despite the name change. See here:

https://bbs.archlinux.org/viewtopic.php?pid=1221889#p1221889

**Comment by caarlos0** [comment]
ok, thanks @amatriain.

On Thu, Jan 24, 2013 at 7:06 AM, amatriain notifications@github.com wrote:

> According to an Arch Linux dev, rmagick should improve its imagemagick
> detection routines, so that it can find the imagemagick libraries despite
> the name change. See here:
> 
> https://bbs.archlinux.org/viewtopic.php?pid=1221889#p1221889
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/rmagick/rmagick/issues/60#issuecomment-12642713.

## 

Atenciosamente,
_Carlos Alexandro Becker_
http://caarlos0.github.com/about

**Comment by C404** [comment]
Thx @jedschneider it did the job perfectly !

**Comment by filipechagas** [comment]
The solution presented by djudd and jedschneider have worked like a charm =)

**Comment by dimsuz** [comment]
Still the proper way would be for rmagick gem to use 'Magick-config' scripts to get location of libraries, rather than hardcoding them.

Unfortunately I am not able to help, because I know nothing about ruby gems internal kitchen and trying to figure that out would take a long time for me, rather than this being a few minutes work for an expert (I guess) :)

**Comment by jedschneider** [comment]
rmagick, afaik, is no longer maintained to any degree. after this issue we switched our projects to mini_magick. I poked around in the internals for about an hour trying to fix this issue without any real success.

**Comment by arthurnn** [comment]
@jedschneider same in here.

**Comment by poiati** [comment]
Try this: http://blog.paulopoiati.com/2013/01/28/installing-rmagick-in-mac-os-x-mountain-lion-with-homebrew/

**Comment by thicksoup** [comment]
@polati A simpler way is to just do:
brew switch imagemagick 6.7.7-6

**Comment by caarlos0** [comment]
anyway, there is another OS out there, not just OSX.

When will we have some fix?

On Tue, Jan 29, 2013 at 10:29 AM, thicksoup notifications@github.comwrote:

> @polati A simpler way is to just do:
> brew switch imagemagick 6.7.7-6
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/rmagick/rmagick/issues/60#issuecomment-12832380.

## 

Atenciosamente,
_Carlos Alexandro Becker_
http://caarlos0.github.com/about

**Comment by filipechagas** [comment]
The last commit to this project is from 2 years ago. I don't think we're going to have any fix on this.
I suggest you to move onto https://github.com/minimagick/minimagick

![Screen Shot 2013-01-29 at 10 43 42](https://f.cloud.github.com/assets/131353/106888/d2fe8ba2-6a11-11e2-8891-9c402ebeee82.png)

**Comment by caarlos0** [comment]
sounds good enought. Thanks.

On Tue, Jan 29, 2013 at 10:46 AM, Filipe Chagas notifications@github.comwrote:

> The last commit to this project is from 2 years ago. I don't think we're
> going to have any fix on this.
> I suggest you to move onto https://github.com/minimagick/minimagick
> 
> [image: Screen Shot 2013-01-29 at 10 43 42]https://f.cloud.github.com/assets/131353/106888/d2fe8ba2-6a11-11e2-8891-9c402ebeee82.png
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/rmagick/rmagick/issues/60#issuecomment-12832908.

## 

Atenciosamente,
_Carlos Alexandro Becker_
http://caarlos0.github.com/about

**Comment by skalee** [comment]
This pull request by @jo fixed the issue for me: https://github.com/rmagick/rmagick/pull/63

**Comment by mmaiza** [comment]
I do not have a mac and did not test this but it should be fixed in the new release 2.13.2
